lwconfig/config.go

```diff
@@ -0,0 +1,140 @@
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/BurntSushi/toml"
+	"github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+)
+
+// Profiles is the representation of the $HOME/.lacework.toml
+//
+// Example:
+//
+// [default]
+// account = "example"
+// api_key = "EXAMPLE_0123456789"
+// api_secret = "_0123456789"
+//
+// [dev]
+// account = "dev"
+// api_key = "DEV_0123456789"
+// api_secret = "_0123456789"
+//
+// [prod]
+// account = "coolcorp"
+// subaccount = "prod-business"
+// api_key = "PROD_0123456789"
+// api_secret = "_0123456789"
+// version = 2
+type Profiles map[string]Profile
+
+// Profile represents a single profile within a configuration file
+type Profile struct {
+	Account    string `toml:"account"`
+	Subaccount string `toml:"subaccount,omitempty"`
+	ApiKey     string `toml:"api_key" survey:"api_key"`
+	ApiSecret  string `toml:"api_secret" survey:"api_secret"`
+	Version    int    `toml:"version,omitzero"`
+}
+
+// Verify will return an error is there is one required setting missing
+func (p *Profile) Verify() error {
+	if p.Account == "" {
+		return errors.New("account missing")
+	}
+	if p.ApiKey == "" {
+		return errors.New("api_key missing")
+	}
+	if p.ApiSecret == "" {
+		return errors.New("api_secret missing")
+	}
+	return nil
+}
+
+// DefaultConfigPath returns the default path where the Lacework config file
+// is located, which is at $HOME/.lacework.toml
+func DefaultConfigPath() (string, error) {
+	home, err := homedir.Dir()
+	if err != nil {
+		return "", err
+	}
+	return path.Join(home, ".lacework.toml"), nil
+}
+
+// LoadProfiles loads all the profiles from the default location ($HOME/.lacework.toml)
+func LoadProfiles() (Profiles, error) {
+	configPath, err := DefaultConfigPath()
+	if err != nil {
+		return Profiles{}, err
+	}
+
+	return LoadProfilesFrom(configPath)
+}
+
+// LoadProfilesFrom loads all the profiles from the provided location
+func LoadProfilesFrom(configPath string) (Profiles, error) {
+	if configPath == "" {
+		return Profiles{}, errors.New("unable to load profiles. Specify a configuration file.")
+	}
+
+	var profiles Profiles
+	if _, err := toml.DecodeFile(configPath, &profiles); err != nil {
+		return profiles, errors.Wrap(err, "unable to decode profiles from config")
+	}
+
+	return profiles, nil
+}
+
+// StoreProfileAt updates a single profile from the provided configuration file
+func StoreProfileAt(configPath, name string, profile Profile) error {
+	cPath := configPath
+	if cPath == "" {
+		defaultPath, err := DefaultConfigPath()
+		if err != nil {
+			return err
+		}
+		cPath = defaultPath
+	}
+
+	var (
+		profiles = Profiles{}
+		err      error
+	)
+	if _, err = os.Stat(cPath); err == nil {
+		if profiles, err = LoadProfilesFrom(cPath); err != nil {
+			return err
+		}
+	}
+
+	profiles[name] = profile
+	var buf = new(bytes.Buffer)
+	if err = toml.NewEncoder(buf).Encode(profiles); err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(cPath, buf.Bytes(), 0600)
+}
```

Comment:
Is the assignment to cPath needed here?
```suggestion
		if configPath == "" {
		defaultPath, err := DefaultConfigPath()
		if err != nil {
			return err
		}
		configPath = defaultPath
	}

	var (
		profiles = Profiles{}
		err      error
	)
	if _, err = os.Stat(configPath); err == nil {
		if profiles, err = LoadProfilesFrom(configPath); err != nil {
			return err
		}
	}

	profiles[name] = profile
	var buf = new(bytes.Buffer)
	if err = toml.NewEncoder(buf).Encode(profiles); err != nil {
		return err
	}

	return ioutil.WriteFile(configPath, buf.Bytes(), 0600)
```